layouts/partials/json-nav-item.html

```diff
@@ -21,7 +21,10 @@
 {{ range .NavPage.Data.Pages}}
     {{ $children = $children | append (partial "json-nav-item" (dict "CurrentPage" $currentPage "NavPage" . "Level" $childLevel )) }}
 {{ end }}
-{{ $base := dict "current" .CurrentPage.File.Filename "type" $type "id" .NavPage.File.Filename "title" .NavPage.Title "level" .Level "collapsed" false  "path" .NavPage.File.Filename "url" $link "roles" (slice (.NavPage.Params.Roles | default "All Roles")) "scope" (slice) "slug" .NavPage.Params.Slug }}
+{{ $base := dict "type" $type "id" .NavPage.File.Filename "title" .NavPage.Title "level" .Level "collapsed" false  "path" .NavPage.File.Filename "url" $link "roles" (slice (.NavPage.Params.Roles | default "All Roles")) "scope" (slice) "slug" .NavPage.Params.Slug  }}
+{{ with .CurrentPage.File }}
+    {{ $base = $base | append (dict "current" .Filename) }}
+{{ end }}
```

Comment:
I feel like appending is overkill. Just setting a var and then adding it will be simpler. I also noticed when generating that it either wasn't appending correctly or because it was blank it was omitting the field. Something that probably shouldn't happen in case something on the backend relies on that field being present
```
{{ $currentFilename := "" }}
{{ with .CurrentPage.File }}
    {{ $currentFilename = .Filename }}
{{ end }}
{{ $base := dict "current" $currentFilename "type" $type...
```

Reply:
Fair enough